### ISSUE TITLE
[BUG] NPM is updating nested module's dependencies

### ISSUE THREAD

**Issue by arielnmz** [npm/cli#1953]
### Current Behavior:
Whenever I install a "top-level" module that happens to be a dependency of another module, npm also updates the nested dependency. Example:

```
/ Project A
|_  Library v5
|_  Project B
    |_  Library v3
```

This is Project A's package.json:

```
{
  "dependencies": {
    "library": "git+ssh://git@bitbucket.org/me/library.git#v5"
    "project-b": "git+ssh://git@bitbucket.org/me/project-b.git"
  }
}
```

And this is Project B's package.json:

```
{
  "dependencies": {
    "library": "git+ssh://git@bitbucket.org/me/library.git#v3"
  }
}
```

If I install Project B first, the package-lock.json file says that Project B depends on Library#v3. But if I then do `npm i project-a` which depends on Library#v5, npm ALSO updates Project B's dependency, so the package-lock.json ends up like this:

```
/ Project A
|_  Library v5
|_  Project B
    |_  Library v5 <-- This should NOT happen, right?
```

### Expected Behavior:
I expect NPM to leave Project B's dependencies alone, and not update or change them

### Steps To Reproduce:
1. Install Project B in Project A
2. Install Library in Project A
3. The package-lock.json now says that both Project A and Project B depend on the same Library (which they don't)

### Environment:
Windows 10 Pro
Node LTS/Dubnium (v10.22)
NPM v6.14

**Comment by mcclenney** [comment]
What version of NPM v6.14 are you on? I had this problem, https://github.com/npm/cli/issues/1400, with git dependencies starting in v6.14.5, but it was fixed in v6.14.7 with commit 36e6c01d334c4db75018bc6a4a0bef726fd41ce4

**Comment by arielnmz** [comment]
I don't recall, but I did update to the latest patch of dubnium and it went away
